core/src/main/scala/org/http4s/HttpDate.scala

```diff
@@ -155,6 +155,8 @@ object HttpDate {
     val dayName =
       List("Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun")
         .map(string1)
+        .zipWithIndex
+        .map { case (s, i) => string1(s).as(i) }
```

Comment:
This didn't really need to be here.  It does if we tighten up weekday validation.